programs/pkey/ecdh_curve25519.c

```diff
@@ -206,13 +190,13 @@ int main( int argc, char *argv[] )
     /*
      * Verification: are the computed secrets equal?
      */
-    mbedtls_printf( "  . Checking if both computed secrets are equal..." );
+    mbedtls_printf( "  . Check if both calculated secrets are equal..." );
     fflush( stdout );
 
-    ret = mbedtls_mpi_cmp_mpi( &ctx_cli.MBEDTLS_PRIVATE(z), &ctx_srv.MBEDTLS_PRIVATE(z) );
+    ret = memcmp( secret_srv, secret_cli, sizeof( secret_srv ) );
```

Comment:
My concern here is that neither secret_srv not secret_cli are initialised, thus if the ouput secret is not the size of the buffer, then this could fail.

I would be inclined to zero the two buffers, and also check the two olens (from calc_secret above) are the same here, and use that size rather than sizeof buffer.